src/main/viewManager.js

```diff
@@ -23,41 +23,45 @@ export class ViewManager {
     this.urlView = null;
   }
 
+  loadServer = (server, mainWindow) => {
+    const srv = new MattermostServer(server.name, server.url);
+    const view = new MattermostView(srv, mainWindow, this.viewOptions);
+    this.views.set(server.name, view);
+    view.setReadyCallback(this.activateView);
+    view.load();
+    view.on(UPDATE_TARGET_URL, this.showURLView);
+  }
+
   // TODO: we shouldn't pass the main window, but get it from windowmanager
   // TODO: we'll need an event in case the main window changes so this updates accordingly
   load = (mainWindow) => {
-    this.configServers.forEach((server) => {
-      const srv = new MattermostServer(server.name, server.url);
-      const view = new MattermostView(srv, mainWindow, this.viewOptions);
-      this.views.set(server.name, view);
-      view.setReadyCallback(this.activateView);
-      view.load();
-      view.on(UPDATE_TARGET_URL, this.showURLView);
-    });
+    this.configServers.forEach((server) => this.loadServer(server, mainWindow));
   }
 
   reloadConfiguration = (configServers, mainWindow) => {
     this.configServers = configServers;
     const oldviews = this.views;
     this.views = new Map();
-    this.load(mainWindow);
-
-    let newView = null;
-
-    //TODO: think of a more gradual approach, this would reload all tabs but in most cases they will be the same or have small variation
-    for (const view of oldviews.values()) {
-      // try to restore view to the same tab if possible, but if not use initial one
-      if (view.isVisible) {
-        log.info(`will try to restore ${view.name}`);
-        newView = this.views.get(view.name);
+    const sorted = this.configServers.sort((a, b) => a.order - b.order);
+    let setFocus;
+    sorted.forEach((server) => {
+      const recycle = oldviews.get(server.name);
+      if (recycle) {
```

Comment:
Would it be worth checking parameters here like the URL? I'm not sure if the tab would load a new URL here if it was changed in the settings. Unless we're manually kicking it on the save.

Reply:
fixed